cmd/notation/internal/outputs/inspect.go

```diff
@@ -0,0 +1,235 @@
+	sigEnvelope, err := signature.ParseEnvelope(mediaType, sigFile)
+	skippedSignatures := false
+	if err != nil {
+		logSkippedSignature(digest, err)
+		skippedSignatures = true
+		return nil, nil
+	}
+
+	envelopeContent, err := sigEnvelope.Content()
+	if err != nil {
+		logSkippedSignature(digest, err)
+		skippedSignatures = true
+		return nil, nil
+	}
+
+	signedArtifactDesc, err := envelope.DescriptorFromSignaturePayload(&envelopeContent.Payload)
+	if err != nil {
+		logSkippedSignature(digest, err)
+		skippedSignatures = true
+		return nil, nil
+	}
+
+	signatureAlgorithm, err := proto.EncodeSigningAlgorithm(envelopeContent.SignerInfo.SignatureAlgorithm)
+	if err != nil {
+		logSkippedSignature(digest, err)
+		skippedSignatures = true
+		return nil, nil
+	}
+
+	sig := SignatureOutput{
+		MediaType:             mediaType,
+		Digest:                digest,
+		SignatureAlgorithm:    string(signatureAlgorithm),
+		SignedAttributes:      getSignedAttributes(InspectOutput{}.outputFormat, envelopeContent),
+		UserDefinedAttributes: signedArtifactDesc.Annotations,
+		UnsignedAttributes:    getUnsignedAttributes(envelopeContent),
+		Certificates:          getCertificates(InspectOutput{}.outputFormat, envelopeContent),
+		SignedArtifact:        *signedArtifactDesc,
+	}
+
+	// clearing annotations from the SignedArtifact field since they're already
+	// displayed as UserDefinedAttributes
+	sig.SignedArtifact.Annotations = nil
+
+	output.Signatures = append(output.Signatures, sig)
+
+	if skippedSignatures {
+		return errors.New("at least one signature was skipped and not displayed"), nil
+	}
+	return nil, output.Signatures
+}
+
+func PrintOutput(outputFormat string, ref string, output InspectOutput) error {
+	if outputFormat == cmd.OutputJSON {
+		return ioutil.PrintObjectAsJSON(output)
+	}
+
+	if len(output.Signatures) == 0 {
+		fmt.Printf("%s has no associated signature\n", ref)
+		return nil
+	}
+
+	fmt.Println("Inspecting all signatures for signed artifact")
+	root := tree.New(ref)
+
+	for _, signature := range output.Signatures {
+		sigNode := root
+		if !(signature.MediaType == "jws" || signature.MediaType == "cose") {
+			cncfSigNode := root.Add(registry.ArtifactTypeNotation)
+			sigNode = cncfSigNode.Add(signature.Digest)
+		}
+		sigNode.AddPair("media type", signature.MediaType)
+		sigNode.AddPair("signature algorithm", signature.SignatureAlgorithm)
+
+		signedAttributesNode := sigNode.Add("signed attributes")
+		addMapToTree(signedAttributesNode, signature.SignedAttributes)
+
+		userDefinedAttributesNode := sigNode.Add("user defined attributes")
+		addMapToTree(userDefinedAttributesNode, signature.UserDefinedAttributes)
+
+		unsignedAttributesNode := sigNode.Add("unsigned attributes")
+		addMapToTree(unsignedAttributesNode, signature.UnsignedAttributes)
+
+		certListNode := sigNode.Add("certificates")
+		for _, cert := range signature.Certificates {
+			certNode := certListNode.AddPair("SHA256 fingerprint", cert.SHA256Fingerprint)
```

Comment:
I have a use case that the user would like to have an array of SHA256 fingerprints, starting from the leaf cert to the root cert, in the output. For example, if the SHA256 fingerprints for leaf is `aaaaa`, for intermediate 1 is `bbbbb`, for intermediate 2 is `ccccc`, and for root is `ddddd`. Then when I run `notation blob inspect my_sig`, I could get an array `["aaaaa", "bbbbb", "ccccc", "ddddd"]` from the final output. 
The current implementation does not support it, where I have to copy/paste/assemble such an array myself.
/cc: @yizha1 @FeynmanZhou

Reply:
Displaying duplicate information in CLI output is not recommended. Instead, users can opt to emit notation output in JSON format and utilize tools like jq to extract the desired text.

Reply:
> Displaying duplicate information in CLI output is not recommended. Instead, users can opt to emit notation output in JSON format and utilize tools like jq to extract the desired text.

Asking users to do a `jq` on the output is extra work though, where we could save this effort for them by adding a field in the output. @yizha1 for your comments please.

Reply:
@priteshbandi Here is the scenario:

Users produce OCI signatures based on Notary Project specifications. When users attach signatures to OCI artifact in OCI registries, users need to add a specific annotation `io.cncf.notary.x509chain.thumbprint#S256`  per [Notary Project OCI signature spec](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#oci-signatures).  It is not easy for users to 1) generate the fingerprints, 2) figure out the order based on the spec. 

As `blob inspect` has all these information, so it will improve the user experience to get ordered x509 chain thumbprints in an easy way, and help more users to adopt Notary Project spec. So, my suggestion would be `blob inspect` to support output certificate SHA256 fingerprints with a new field under current `certificates`, for example

certificate SHA256 fingerprints: ["leaf-sha256-fingerprint","intermediate-sha256-fingerprint","root-sha256-fingerprint"]

This will not block this PR, if you agree, we can create a new issue to track this work. It will be a simple work, but save a big effort for our users.